indico/modules/events/persons/util.py

```diff
@@ -0,0 +1,71 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2017 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from indico.modules.events.models.persons import EventPerson
+from indico.modules.users import User
+from indico.modules.users.models.users import UserTitle
+from indico.util.i18n import _
+from indico.util.user import principal_from_fossil
+
+
+def create_event_person(event, create_untrusted_persons=False, **data):
+    """Create an event person from data passed as kwargs."""
+    title = next((x.value for x in UserTitle if data.get('title') == x.title), None)
+    return EventPerson(event=event, email=data.get('email', '').lower(), _title=title,
+                       first_name=data.get('firstName'), last_name=data['familyName'],
+                       affiliation=data.get('affiliation'), address=data.get('address'),
+                       phone=data.get('phone'), is_untrusted=create_untrusted_persons)
+
+
+def get_event_person_for_user(event, user, create_untrusted_persons=False):
+    """Return the event person that links to a given User/Event (if any)."""
+    return EventPerson.for_user(user, event, is_untrusted=create_untrusted_persons)
+
+
+def get_event_person(event, data, create_untrusted_persons=False, allow_external=False, allow_emails=False,
+                     allow_networks=False):
+    """
+    Get an EventPerson from dictionary data.
```

Comment:
should be in the `"""` line since it's a short single-line description

Reply:
There's more stuff below. The first sentence should always be 1 line only IIRC.

Reply:
> The summary line may be on the same line as the opening quotes or on the next line.

so yeah, it's not required - but consistent with our other docstrings ;)

Reply:
Well, I tend to like these two styles:

```
"""This function doesn't do anything."""
```

```
"""
This function doesn't do anything.

It's pretty useless indeed, and it returns a random value just to
confuse you.
"""
```

It annoys me a bit the lack of consistency of our current approach:

```
"""This function doesn't do anything.

It's pretty useless indeed, and it returns a random value just to
confuse you.
"""
```
But yeah, that's what we use mostly.